client/web/src/components/legacy.tsx

```diff
@@ -0,0 +1,271 @@
+import React from "react"
+import useNodeData from "src/hooks/node-data"
+
+// legacy.tsx contains a set of components that (crudely) implement the pre-2023
+// web client. These are implemented purely to ease migration to the new
+// React-based web client, and will eventually be completely removed.
+
+export function Header() {
+  const data = useNodeData()
```

Comment:
Rather than calling `useNodeData` in all of these components, we could call it once in `App.tsx` and pass it as props to each of the components. That might also help us keep things in sync once we make mutations - so all data loads together.

Reply:
done